transformer_engine/pytorch/attention.py

```diff
@@ -1280,6 +1345,9 @@ def forward(
         cu_seqlens_q_per_step = [None for _ in range(cp_size)]
         cu_seqlens_kv_per_step = [None for _ in range(cp_size)]
 
+        assert qkv_format == "thd" or (
+            q.shape[seq_dim] % 2 == 0 and k.shape[seq_dim] % 2 == 0
+        ), "Sequence length per GPU needs to be divisible by 2!"
```

Comment:
Is the requirement that the sequence is divisible by 2 or by 2 * cp_size?

Reply:
Should be divisible by 2. This is an assert for the sequence chunk length in each GPU, not an assert for full sequence.